skopt/tree_opt.py

```diff
@@ -33,8 +33,15 @@ def _tree_minimize(func, dimensions, base_estimator, maxiter,
         raise ValueError("Total number of iterations set by maxiter has to"
                          " be larger or equal to n_start.")
 
-    Xi = space.rvs(n_samples=n_start, random_state=rng)
-    yi = [func(x) for x in Xi]
+    if x0 is None:
+        x0 = space.rvs(n_samples=n_start, random_state=rng)
```

Comment:
This assumes that if `x0` is provided, `n_start` is not useful, right? I would go with your logic in `dummy_opt.py` instead. WDYT?